### ISSUE TITLE
Problems with enums which are named the same as the register

### ISSUE THREAD

**Issue by odinthenerd** [kvasir-io/HardwareDescriptionFileGenerator#1]
one thing the current parser is doing wrong is the special case where a register has the same name as one of its enum values. Here is a code example 
```
        ///Write '1' to Enable interrupt on EVENTS_DISABLED event
        enum class DisabledVal : unsigned {
            disabled=0x00000000,     ///<Read: Disabled
            enabled=0x00000001,     ///<Read: Enabled
            set=0x00000001,     ///<Enable
        };
        constexpr Register::FieldLocation<Addr,Register::maskFromRange(4,4),Register::ReadWriteAccess,DisabledVal> disabled{}; 
        namespace DisabledValC{
            constexpr Register::FieldValue<decltype(disabled)::Type,DisabledVal::disabled> disabled{};
            constexpr Register::FieldValue<decltype(disabled)::Type,DisabledVal::enabled> enabled{};
            constexpr Register::FieldValue<decltype(disabled)::Type,DisabledVal::set> set{};
        }
```
here `disabled` is both an enum value and a register name so we need to namespace qualify it.

**Comment by jackieokay** [comment]
Would you prefer to get rid of the namespace around the constexpr field values values and disambiguate the names another way? I think that the constexpr field values should have the original field name (disabled) and the enum name could be e.g. disabled_val

I find it strange that the compiler can't disambiguate the enum name and the field value name, since the enum value name has to be qualified with the enum class name. It seems like that namespace shouldn't be needed.

**Comment by odinthenerd** [comment]
yes that might be a good idea. So code would look like `apply(write(Radio::disabled_val));` or `if(field_equals(Radio::disabled_val))`. I initially used a namespace so that people could pull it in with a `using namespace` but the more I look at the naming conventions I think there is not enough information left for that to be a good design pattern
